File: 2511. Maximum Enemy Forts That Can Be Captured/Solution.py
from typing import List


class Solution:
    def captureForts(self, forts: List[int]) -> int:
        pre, res = -1, 0
        for i, e in enumerate(forts):
            if e != 0:
                if pre > -1 and e != forts[pre]:
                    res = max(res, i - pre - 1)
                pre = i
        return res

File: 2511. Maximum Enemy Forts That Can Be Captured/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int captureForts(vector<int>& forts)
    {
        int pre = -1, res = 0, n = forts.size();
        for (int i = 0; i < n; i++) {
            if (forts[i] != 0) {
                if (pre > -1 && forts[i] != forts[pre])
                    res = max(res, i - pre - 1);
                pre = i;
            }
        }
        return res;
    }
};
